source/agora/node/Ledger.d

```diff
@@ -1145,14 +1145,20 @@ public class ValidatingLedger : Ledger
     public void prepareNominatingSet (out ConsensusData data, ulong max_txs)
         @safe
     {
+        ulong rounded (ulong offset)
+        {
+            // round down to the block interval or 1 min if block interval in more than 1 min
+            return offset - (offset % min(60, this.params.BlockInterval.total!"seconds"));
```

Comment:
`ConsensusParams` should be an absolute. Instead of patching it here, we should error out if the value cannot be respected. Currently we have a block interval of 1 in the test, which is why the test failed before.

Reply:
why do we do rounding here?

Reply:
I have dropped the second commit as it is not fixing the problem we have and it needs some more design.